File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let maxWidth = 0, maxHeight = 0;
    sizes.map(([w, h]) => [Math.max(w, h), Math.min(w, h)])
         .forEach(([w, h]) => {
             if (w > maxWidth) maxWidth = w;
             if (h > maxHeight) maxHeight = h;
         })
    return maxWidth * maxHeight;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    for i in range(len(sizes)):
        [w, h] = sizes[i]
        if w < h:
            w, h = h, w
        sizes[i] = [w, h]
    
    max_w = max(w for w, h in sizes)
    max_h = max(h for w, h in sizes)
    
    return max_w * max_h